strawberry_django/auth/utils.py

```diff
@@ -0,0 +1,23 @@
+from strawberry.types import Info
+
+
+def get_current_user(info: Info):
+    """Get and return the current user based on various scenarios."""
+    try:
+        user = info.context.request.user
+    except AttributeError:
+        try:
+            # When running queries/mutations in ASGI mode, the user is moved into the consumer scope
+            user = info.context.get("request").consumer.scope["user"]
+        except AttributeError:
+            # When using this through websockets / subscriptions, scope sits inside of the request
+            user = info.context.get("request").scope.get("user")
+
+    # Access an attribute inside the user object to force loading it in async contexts.
+    user.is_authenticated
```

Comment:
Also, `user` might be `None` in here
```suggestion
    if user is not None:
        user.is_authenticated
```